### PR TITLE
*: reset statement context for prepared statements

### LINKED ISSUE #3381: `insert` prepared statement is not consistent with MySQL
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
``` sql
create table t (a char(1));
prepare stmt from 'insert into t value("123")';
execute stmt;
```

2. What did you expect to see?
``` sql
mysql> execute stmt;
ERROR 1406 (22001): Data too long for column 'a' at row 1
```

3. What did you see instead?
``` sql
mysql> execute stmt;
Query OK, 1 row affected (0.00 sec)
```

4. What version of TiDB are you using (`tidb-server -V`)?
Git Commit Hash: 62450b902432f70848d315311d6b3d79325c72c9
UTC Build Time:  2017-06-04 12:22:25

### PR DESCRIPTION
**Pull request by XuHuaiyu**
fix #3381 
PTAL @coocood @shenli

### CODE REVIEW

**Comment by coocood** [comment]
Please fix CI

**Comment by coocood** [comment]
LGTM

**Comment by XuHuaiyu** [comment]
PTAL @shenli @hanfei1991

**Comment by ngaut** [comment]
LGTM

File: executor/prepared.go [hunk 1]
```diff
@@ -325,3 +326,39 @@ func CompileExecutePreparedStmt(ctx context.Context, ID uint32, args ...interfac
...
 }
+
+// ResetStmtCtx resets the StmtContext.
+// Before every execution, we must clear statement context.
+func ResetStmtCtx(ctx context.Context, s ast.StmtNode) {
+	sessVars := ctx.GetSessionVars()
+	sc := new(variable.StatementContext)
+	switch s.(type) {
+	case *ast.UpdateStmt, *ast.InsertStmt, *ast.DeleteStmt:
+		sc.IgnoreTruncate = false
```
Comment (shenli) [inline executor/prepared.go]: For a new variable.StatementContext, sc.IgnoreTruncate is already false.

Comment (coocood) [inline executor/prepared.go]: This PR just move the old code to another package.
We better not moving code and modify it at the same time.

Comment (shenli) [inline executor/prepared.go]: ok

**Pull request merged**